s2/bq/doc/REFERENCE.md

```diff
@@ -0,0 +1,162 @@
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners given a longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ID_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Convert from hilbert quadtree id to s2 cell id.
+
+* `key`: `STRING` quadkey to be converted.
+
+#### ID_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the s2 cell id for a given longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### KEY_FROMID
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Convert from s2 cell id to hilbert quadtree id.
+
+* `id`: `INT64` s2 cell id to be converted.
+
+#### KEY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the quadkey for a given longitude, latitude and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### LONG_FROMID
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a give s2 cell id.
+
+* `id`: `INT64` s2 cell id.
+
+#### LONG_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a quadkey.
+
+* `key`: `STRING` quadkey.
+
+#### LAT_FROMID
+
+{{% bannerNote type="code" %}}
+s2.LAT_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Extract the latitude component from a give s2 cell id.
+
+* `id`: `INT64` s2 cell id.
+
+#### LAT_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.LAT_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Extract the latitude component from a quadkey.
+
+* `key`: `STRING` quadkey.
+
+#### LONGLAT_ASS2
+
+{{% bannerNote type="code" %}}
+s2.LONGLAT_ASS2(longitude FLOAT64, latitude FLOAT64, resolution NUMERIC)
+{{%/ bannerNote %}}
+
+Returns the s2 representation for a given level of detail and geographic coordinates.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ST_ASS2
+
+{{% bannerNote type="code" %}}
+s2.ST_ASS2(point GEOGRAPHY, resolution NUMERIC)
+{{%/ bannerNote %}}
+
+Converts a given point at given level of detail to a s2 cell id.
+
+* `point`: `GEOGRAPHY` point we want to get the quadint from.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ST_GEOGFROMLONGLAT_BOUNDARY
```

Comment:
Again, lots of duplicated functionality here. We should start simple and build from there, this just adds a lot of complexity to something that should be as simple as I have a geography and I want the s2 cell id, or I have the s2 cell id and I want the geography.